include/deal.II/base/mpi.templates.h

```diff
@@ -95,21 +95,23 @@ namespace Utilities
 #ifdef DEAL_II_WITH_MPI
         if (job_supports_mpi())
           {
-            MPI_Allreduce (values != output
-                           ?
-                           // TODO This const_cast is only needed for older
-                           // (e.g., openMPI 1.6, released in 2012)
-                           // implementations of MPI-2. It is not needed as of
-                           // MPI-3 and we should remove it at some point in
-                           // the future.
-                           const_cast<void *>(static_cast<const void *>(values))
-                           :
-                           MPI_IN_PLACE,
-                           static_cast<void *>(output),
-                           static_cast<int>(size),
-                           internal::mpi_type_id(values),
-                           mpi_op,
-                           mpi_communicator);
+            const int ierr = MPI_Allreduce
+                             (values != output
+                              ?
+                              // TODO This const_cast is only needed for older
+                              // (e.g., openMPI 1.6, released in 2012)
+                              // implementations of MPI-2. It is not needed as
+                              // of MPI-3 and we should remove it at some
+                              // point in the future.
+                              const_cast<void *>(static_cast<const void *>(values))
+                              :
+                              MPI_IN_PLACE,
+                              static_cast<void *>(output),
+                              static_cast<int>(size),
+                              internal::mpi_type_id(values),
+                              mpi_op,
+                              mpi_communicator);
+            AssertThrow(ierr == MPI_SUCCESS, ExcMPI(ierr));
```

Comment:
You could make all of these checks simpler if you had a macro that expands to this very line and only takes `ierr` add argument. I think we have such variations of Assert in exceptions.h for other repetitive cases.